axelrod/strategies/axelrod_second.py

```diff
@@ -1305,3 +1305,114 @@ def strategy(self, opponent: Player) -> Action:
             return self.try_return(D, lower_flags=False)
+
+
+class MoreTidemanAndChieruzzi(Player):
+    """
+    Strategy submitted to Axelrod's second tournament by T. Nicolaus Tideman
+    and Paula Chieruzzi (K84R) and came in ninth in that tournament.
+
+    This strategy Cooperates if this player's score exceeds the opponent's
+    score by at least `score_to_beat`.  `score_to_beat` starts at zero and
+    increases by `score_to_beat_inc` every time the opponent's last two moves
+    are a Cooperation and Defection in that order.  `score_to_beat_inc` itself
+    increase by 5 every time the opponent's last two moves are a Cooperation
+    and Defection in that order.
+
+    Additionally, the strategy executes a "fresh start" if the following hold:
+
+    - The strategy would Defect by score (difference less than `score_to_beat`)
+    - The opponent did not Cooperate and Defect (in order) in the last two
+      turns.
+    - It's been at least 10 turns since the last fresh start.  Or since the
+      match started if there hasn't been a fresh start yet.
+
+    A "fresh start" entails two Cooperations and resetting scores,
+    `scores_to_beat` and `scores_to_beat_inc`.
+
+    Names:
+
+    - MoreTidemanAndChieruzzi: [Axelrod1980b]_
+    """
+
+    name = 'More Tideman and Chieruzzi'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {"game"},
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.current_score = 0
+        self.opponent_score = 0
+        self.last_fresh_start = 0
+        self.fresh_start = False
+        self.score_to_beat = 0
+        self.score_to_beat_inc = 0
+
+    def _fresh_start(self):
+        """Give the opponent a fresh start by forgetting the past"""
+        self.current_score = 0
+        self.opponent_score = 0
+        self.score_to_beat = 0
+        self.score_to_beat_inc = 0
+
+    def _score_last_round(self, opponent: Player):
+        """Updates the scores for each player."""
+        # Load the default game if not supplied by a tournament.
+        game = self.match_attributes["game"]
+        last_round = (self.history[-1], opponent.history[-1])
+        scores = game.score(last_round)
+        self.current_score += scores[0]
+        self.opponent_score += scores[1]
+
+    def strategy(self, opponent: Player) -> Action:
+        current_round = len(self.history) + 1
+
+        if current_round == 1:
+            return C
+
+        # Calculate the scores.
+        self._score_last_round(opponent)
+
+        # Check if we have recently given the strategy a fresh start.
+        if self.fresh_start:
+            self._fresh_start()
+            self.last_fresh_start = current_round
+            self.fresh_start = False
+            return C  # Second cooperation
+
```

Comment:
Can you explain the name `opponent_CDd`? I looks like it's just checking that `opponent.history[-2:] == [C, D]` right? Just curious what the final `d` indicates.

Reply:
That's correct.  The second d is because "CDd" is the past tense of "CD."